src/os/structs/nodeutils.js

```diff
@@ -0,0 +1,21 @@
+goog.provide('os.structs.nodeutils');
+
+
+/**
+ * Flattens a tree into a specified array - optionally filtering.
+ * @param {os.structs.ITreeNode} node
+ * @param {Array<os.structs.ITreeNode>} results
+ * @param {function(os.structs.ITreeNode): boolean} opt_filter
+ */
+os.structs.nodeutils.flattenTree = function(node, results, opt_filter) {
+  if (opt_filter ? opt_filter(node) : true) {
+    results.push(node);
+    var children = node.getChildren();
+    if (children) {
+      for (var i = 0; i < node.getChildren().length; i++) {
```

Comment:
This should be `for (var i = 0, n = children.length; i < n; i++)`. We always try to cache the length if it is not changing.

Reply:
Fixed